simba/report.py

```diff
@@ -360,3 +381,259 @@ def write_csv(data: Iterable, file_path, propagate_errors=False):
+    :type schedule: eBus-Toolbox.Schedule
+    """
+    step = 50
+    max_route = int(max([schedule.rotations[rot].distance / 1000 for rot in schedule.rotations]))
+    num_bins = max_route // step + 1  # number of bars in plot
+    labels = [f"{i*step} - {(i+1)*step}" for i in range(num_bins)]
+    # init bins: track distance bins for each vehicle type individually
+    bins = {v_types: [0]*num_bins for v_types in schedule.vehicle_types}
+
+    # fill bins with rotations
+    for rot in schedule.rotations:
+        position = int((schedule.rotations[rot].distance / 1000) // step)
+        bins[schedule.rotations[rot].vehicle_type][position] += 1
+
+    # plot
+    fig, ax = plt.subplots()
+    bar_bottom = [0] * num_bins
+    for v_type in schedule.vehicle_types:
+        ax.bar(labels, bins[v_type], width=0.9, label=v_type, bottom=bar_bottom)
+        for i in range(num_bins):
+            bar_bottom[i] += bins[v_type][i]
+    ax.set_xlabel('Distance [km]')
+    plt.xticks(rotation=30)  # slant labels for better readability
+    ax.set_ylabel('Number of rotations')
+    ax.yaxis.get_major_locator().set_params(integer=True)
+    ax.yaxis.grid(True)
+    ax.set_title('Distribution of bus types over rotation distance')
+    ax.legend()
+    plt.savefig(extended_plots_path / "distribution_bustypes_route_rotations")
+    plt.close()
+
+
+def plot_consumption_per_rotation_distribution(extended_plots_path, schedule):
+    """Plots the distribution of bus types in consumption brackets as a stacked bar chart.
+
+    :param extended_plots_path: directory to save plot to
+    :type extended_plots_path: Path
+    :param schedule: Driving schedule for the simulation, schedule.rotations are used
+    :type schedule: eBus-Toolbox.Schedule
+    """
+
+    step = 50
+    # get number of bins
+    max_con = int(max([schedule.rotations[rot].consumption for rot in schedule.rotations]))
+    num_bins = max_con // step + 1  # number of bars in plot
+    labels = [f"{i*step} - {(i+1)*step}" for i in range(num_bins)]
+    # init bins: track consumption bins for each vehicle type individually
+    bins = {v_types: [0]*num_bins for v_types in schedule.vehicle_types}
+
+    # fill bins with rotations
+    for rot in schedule.rotations:
+        position = int(schedule.rotations[rot].consumption // step)
+        bins[schedule.rotations[rot].vehicle_type][position] += 1
+
+    # plot
+    fig, ax = plt.subplots()
+    bar_bottom = [0] * num_bins
+    for v_type in schedule.vehicle_types:
+        ax.bar(labels, bins[v_type], width=0.9, label=v_type, bottom=bar_bottom)
+        for i in range(num_bins):
+            bar_bottom[i] += bins[v_type][i]
+    ax.set_xlabel('Energy consumption [kWh]')
+    plt.xticks(rotation=30)
+    ax.set_ylabel('Number of rotations')
+    ax.yaxis.get_major_locator().set_params(integer=True)
+    ax.yaxis.grid(True)
+    ax.set_title('Distribution of bus types over rotation energy consumption')
```

Comment:
Same issue with title name as above

Reply:
I would go with "distribution of energy consumption of rotations per vehicle type"